### DISCUSSION TITLE
Transcoding from universal basis ETC1S to ETC2 RGB with punch-through alpha

### DISCUSSION THREAD

**Discussion by limingchina** [KhronosGroup/KTX-Software#477] [Q&A] [answered]
When checking the libktx header file: 
https://github.com/KhronosGroup/KTX-Software/blob/514032b2bc927583f3ec6bddd967dbe2720dc7d3/include/ktx.h

The enum type **ktx_transcode_fmt_e**  only supports ETC2_RGB, ETC2_RGBA. There is no mentioning of ETC_RGB with punch-through alpha.

```
       KTX_TTF_ETC1_RGB = 0,
            /*!< Opaque only. Returns RGB or alpha data, if
                 KTX_TF_TRANSCODE_ALPHA_DATA_TO_OPAQUE_FORMATS flag is
                 specified. */

        KTX_TTF_ETC2_RGBA = 1,
            /*!< Opaque+alpha. EAC_A8 block followed by an ETC1 block. The
                 alpha channel will be opaque for textures without an alpha
                 channel. */
```
Is it difficult to implement the punch-through alpha support? How fast does it compare to the existing ETC1/ETC2 transcoding?

**Comment by lexaknyazev** [comment, marked as answer]
Source RGBA Basis ETC1S data is stored in two separate blocks that are transcoded independently: ETC1S RGB becomes ETC2 RGB (lossless repacking), while ETC1S Alpha becomes ETC2 A8 (simple transcoding). The two transcoded 64-bit blocks are then concatenated into a single ETC2 RGBA 128-bit block.

Producing an ETC2 punch-through alpha blocks would be lossy and noticeably slower because it would require decoding and reencoding ETC1S data.
